### PR TITLE
Basic GDALWarpStacked options support

### PR DESCRIPTION
**Pull request by pomadchin**
- [ ] Test in GPS (@jbouffard help is required here)

### CODE REVIEW

File: vlm/src/main/scala/geotrellis/contrib/vlm/ResampleGrid.scala [hunk 1]
```diff
@@ -1,39 +0,0 @@
...
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- * See the License for the specific language governing permissions and
- * limitations under the License.
- */
-
-package geotrellis.contrib.vlm
-
-import geotrellis.raster.{RasterExtent, GridExtent}
-
-sealed trait ResampleGrid {
```
Comment (echeipesh) [inline vlm/src/main/scala/geotrellis/contrib/vlm/ResampleGrid.scala]: Did this move to `geotrellis.gdal` or some such thing ?

**Comment by echeipesh** [comment]
Related: https://github.com/geotrellis/geotrellis-gdal/pull/50

**Comment by echeipesh** [comment]
`ResampleGrid` should not leave this project really. Furthermore the idea of flattening the Options is much more of an idea for RasterSource than GDAL code itself. I think the .resample/.reproject methods on the `GDALWarpOptions` in this PR form a state machine thats a mirror of `GDALRasterSource` states. Looks like it can be lifted here with protected `reprojectOptions` and `resampleOptions` methods to keep it close and still as clean.

File: vlm/src/main/scala/geotrellis/contrib/vlm/gdal/package.scala [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+      * This method copies gdalwarp -tap logic:
+      *
+      * The actual code reference: https://github.com/OSGeo/gdal/blob/v2.3.2/gdal/apps/gdal_rasterize_lib.cpp#L402-L461
+      * The actual part with the -tap logic: https://github.com/OSGeo/gdal/blob/v2.3.2/gdal/apps/gdal_rasterize_lib.cpp#L455-L461
+      *
+      * The initial PR that introduced that feature in GDAL 1.8.0: https://trac.osgeo.org/gdal/attachment/ticket/3772/gdal_tap.patch
+      * A discussion thread related to it: https://lists.osgeo.org/pipermail/gdal-dev/2010-October/thread.html#26209
+      *
+      */
+    def alignTargetPixels: RasterExtent = {
```
Comment (pomadchin) [inline vlm/src/main/scala/geotrellis/contrib/vlm/gdal/package.scala]: Worrying about this logic here; should it be there at all? However without it we won't get tests won't pass :/

File: vlm/src/main/scala/geotrellis/contrib/vlm/gdal/package.scala [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+    }
+
+    def resample(gridExtent: => GridExtent, resampleGrid: ResampleGrid): GDALWarpOptions = {
+      val rasterExtent = gridExtent.toRasterExtent
+      resampleGrid match {
+        case Dimensions(cols, rows) => self.copy(te = None, cellSize = None, dimensions = (cols, rows).some)
+        case _ =>
+          val re = {
+            val targetRasterExtent = resampleGrid(rasterExtent)
+            if(self.alignTargetPixels) targetRasterExtent.alignTargetPixels else targetRasterExtent
```
Comment (pomadchin) [inline vlm/src/main/scala/geotrellis/contrib/vlm/gdal/package.scala]: It looks like we have to specify targetExtent manually (`-te option`) => to align pixels manually.

File: vlm/src/test/scala/geotrellis/contrib/vlm/RasterSummarySpec.scala [hunk 4]
```diff
@@ -164,17 +167,19 @@ class RasterSummarySpec extends FunSpec with TestEnvironment with BetterRasterMa
       widthCollected shouldBe (widthResampled +- 1e-7)
       heightCollected shouldBe (heightResampled +- 1e-7)
 
-      val Extent(xminc, yminc, xmaxc, ymaxc) = summaryCollected.extent
-      val Extent(xminr, yminr, xmaxr, ymaxr) = summaryResampled.extent
+      // TODO: investigate the reason of why this won't work here
+      // but probably this function should be removed in the future completely and nowhere used
+      // val Extent(xminc, yminc, xmaxc, ymaxc) = summaryCollected.extent
+      // val Extent(xminr, yminr, xmaxr, ymaxr) = summaryResampled.extent
```
Comment (pomadchin) [inline vlm/src/test/scala/geotrellis/contrib/vlm/RasterSummarySpec.scala]: `summaryCollected.extent != summaryResampled.extent` mb a slight error somewhere here. However we don't use this method at all, and other `tileToLayout` tests work and pass.

**Comment by pomadchin** [comment]
@jbouffard can you test this PR with `geopyspark`? I'm worrying about how correct the result output would be.

**Comment by jbouffard** [comment]
@pomadchin Yeah, I can test it now :+1:

**Comment by jbouffard** [comment]
Everything seems to be working okay in GPS :+1:

**Pull request merged**